traits/tests/test_regression.py

```diff
@@ -232,5 +234,28 @@ def test_on_trait_change_with_list_of_extended_names(self):
         self.assertTrue(model.changed)
 
 
+class TestDerivedClassDefaults(unittest.TestCase):
+
+    def test_function_default(self):
+        # Regression test for enthought/traits#411
+
+        # Given
+
+        def fun():
+            pass
+
+        class A(HasTraits):
+            f = Callable()
+
+        class B(A):
+            f = fun
+
+        # When
+        b = B()
+
+        # Then
+        self.assertIsNotNone(b.f)
```

Comment:
Could we go one step further and call `b.f`, and verify that the result is what we expect (`None` in this case, but we could fix up `fun` to return something interesting instead).